processor/src/test/java/org/mapstruct/ap/testutil/runner/CompilingStatement.java

```diff
@@ -586,11 +585,12 @@ public int compare(DiagnosticDescriptor o1, DiagnosticDescriptor o2) {
             if ( result != 0 ) {
                 return result;
             }
-            result = o1.getLine().compareTo( o2.getLine() );
-            if ( result != 0 ) {
-                return result;
+            if ( o1.getLine() != null && o2.getLine() != null ) {
```

Comment:
This is because the messager is `null`, right? So we don't output a line?

Reply:
I keep on having problems with the eclipse lines / jdk lines because of this bug in javac. (annotations reported on the method). So for simple cases I thought we might enable to make line optional. I ran into NPE's, hence this fix.

it has nothing to do with the messager and I can revert if you want. In the end I did not use this, but I left it for the future.

Reply:
I know that bug, that's where there is the `alternativeLine` in the `@Diagnostic`. I've added it so we can workaround that bug

Reply:
ah.. good point.. but still, a null check does not harm here..

Reply:
Yes it doesn't harm, but it would hide the problem when you want to check something by just ignoring it

Reply:
Ok.. I'l remove it..